File: src/com/dengzii/plugin/findview/utils/PsiClassUtils.java
package com.dengzii.plugin.findview.utils;

import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiMethod;

/**
 * <pre>
 * author : dengzi
 * e-mail : denua@foxmail.com
 * github : https://github.com/MrDenua
 * time   : 2019/9/30
 * desc   :
 * </pre>
 */
public class PsiClassUtils {

    public static PsiMethod getMethod(PsiClass psiClass, String methodName) {
        PsiMethod[] methods = getMethods(psiClass);
        for (PsiMethod method : methods) {
            if (method.getName().equals(methodName)) {
                return method;
            }
        }
        return null;
    }

    public static PsiMethod[] getMethods(PsiClass psiClass) {
        return psiClass.getMethods();
    }
}

File: src/main/kotlin/com/dengzii/plugin/findview/utils/PsiClassUtils.kt
package com.dengzii.plugin.findview.utils

import com.intellij.psi.PsiClass
import com.intellij.psi.PsiMethod

/**
 * <pre>
 * author : dengzi
 * e-mail : dengzii@foxmail.com
 * github : [...](https://github.com/dengzii)
 * time   : 2019/9/30
 * desc   :
</pre> *
 */
object PsiClassUtils {
    fun getMethod(psiClass: PsiClass, methodName: String): PsiMethod? {
        val methods = getMethods(psiClass)
        for (method in methods) {
            if (method.name == methodName) {
                return method
            }
        }
        return null
    }

    private fun getMethods(psiClass: PsiClass): Array<PsiMethod> {
        return psiClass.methods
    }
}
